ui/apps/dashboard/src/app/(organization-active)/(dashboard)/env/[environmentSlug]/functions/[slug]/runs/page.tsx

```diff
@@ -51,18 +56,32 @@ export default function RunsPage() {
   const [rawFilteredStatus, setFilteredStatus, removeFilteredStatus] =
     useStringArraySearchParam('filterStatus');
   const [rawTimeField, setTimeField] = useSearchParam('timeField');
+  const [lastDays = '3', setLastDays] = useSearchParam('last');
+
+  const timeField = toTimeField(rawTimeField ?? '') ?? FunctionRunTimeFieldV2.QueuedAt;
+
+  /* TODO: Time params for absolute time filter */
+  // const [fromTime, setFromTime] = useSearchParam('from');
+  // const [untilTime, setUntilTime] = useSearchParam('until');
+
+  /* TODO: When we have absolute time, the start date will be either coming from the date picker or the relative time */
+  const [startTime, setStartTime] = useState<Date>(new Date());
```

Comment:
**suggestion (non-blocking)**: Probably doesn't matter with the TODO here, but will this cause issues b/c `new Date()` will return a new value every time this component renders? I don't know if it needs to be memo'd or we can just forget it for now until the date picker is there?

Reply:
I think `startTime` should be derived (and memoized, like you're suggesting) from the URL state. If the URL has something like `relative=5h` then `startTime` would be 5 hours ago. If the URL has something like `startTime=2024-05-07T11:22:33Z` then it would be `new Date("2024-05-07T11:22:33Z")`.

Also, should we rename `startTime`? It's kinda confusing since it reads like it's related to "started at" when it's really the lower time bound for whichever time field is used (queued, started, or ended)

Reply:
@goodoldneon was making a good point about always inferring the value from the URL, even if they are the initial defaults that users didn't select. That means changing the behaviour so that the defaults in filters are now also part of the URL, and drop the need for this state. I'll make that change either with pagination or the absolute time work